### PR TITLE
Introduce new APIs for avoiding LoginAccount.password

### PR DESCRIPTION
**Pull request by mithrandi**
Also deprecate IUsernameHashedPassword credentials.

Refs #36.

### CODE REVIEW

File: axiom/userbase.py [hunk 1]
```diff
@@ -268,6 +275,33 @@ def addLoginMethod(self, localpart, domain, protocol=ANY_PROTOCOL, verified=Fals
...
 
+    def setPassword(self, newPassword):
+        """
+        Set this account's password unconditionally.
+
+        @param newPassword: The new password.
+
+        @return: A deferred firing when the password has been changed.
+        """
+        self.password = unicode(newPassword)
```
Comment (jonathanj) [inline axiom/userbase.py]: I think we need to handle `None` specially here. See `LoginBase.addAccount`.

**Comment by glyph** [comment]
Any chance @mithrandi will finish this now? :)

**Comment by mithrandi** [comment]
It's still on my radar :smile_cat:

**Comment by glyph** [comment]
@mithrandi 

"ping"

**Comment by glyph** [comment]
(Still really hoping this eventually gets landed...)

**Comment by glyph** [comment]
@jonathanj Is the `None` thing your only feedback?  Would it be good to land without it?

**Comment by jonathanj** [comment]
@glyph Yes, I think so. I'm okay to see it merged without that though.

File: axiom/userbase.py (hunk 1 above)
Comment (mithrandi) [reply axiom/userbase.py]: Done.

**Comment by glyph** [comment]
OH MY GOD AMAZING, IT'S FINALLY HAPPENING

EVERYBODY GO PUT AXIOM IN PRODUCTION RIGHT NOW

**Comment by glyph** [comment]
Thank you so much @mithrandi @jonathanj !

**Comment by glyph** [comment]
Did this ever get into a PyPI release?

**Comment by mithrandi** [comment]
I think we still need to roll a new release.

**Comment by mithrandi** [comment]
FTR 0.8.0 was released with these changes; need to land #107 now.

**Pull request merged**